src/main/java/io/reactivex/Flowable.java

```diff
@@ -7669,9 +7669,24 @@ public final void blockingSubscribe(Subscriber<? super T> subscriber) {
     }
 
     /**
-     * Returns a Flowable that emits all items emitted by the source Publisher that are distinct.
+     * Returns a Flowable that emits all items emitted by the source Publisher that are distinct
+     * based on {@link Object#equals(Object)} comparison.
      * <p>
      * <img width="640" height="310" src="https://raw.github.com/wiki/ReactiveX/RxJava/images/rx-operators/distinct.png" alt="">
+     * <p>
+     * It is recommended the elements' class {@code T} in the flow overrides the default {@code Object.equals()} to provide
```

Comment:
Since we use `HashSet` internally, we should probably recommend properly overriding `hasCode()` as well

Reply:
Same below